src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -465,37 +466,61 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         // constraints we have accumulated.
         let mut inferred_values = self.liveness_constraints.clone();
 
-        while changed {
-            changed = false;
-            debug!("propagate_constraints: --------------------");
-            for constraint in &self.constraints {
-                debug!("propagate_constraints: constraint={:?}", constraint);
-
-                // Grow the value as needed to accommodate the
-                // outlives constraint.
-                let Ok(made_changes) = self.dfs(
-                    mir,
-                    CopyFromSourceToTarget {
-                        source_region: constraint.sub,
-                        target_region: constraint.sup,
-                        inferred_values: &mut inferred_values,
-                        constraint_point: constraint.point,
-                        constraint_span: constraint.span,
-                    },
-                );
+        let dependency_map = self.build_dependency_map();
+        let mut dirty_list: Vec<_> = (0..self.constraints.len()).collect();
+        let mut dirty_bit_vec = BitVector::new(dirty_list.len());
```

Comment:
Nit: this starts as all zeroes, but we really want it to start as all ones I suppose. But it won't be *wrong* this way, just mildly less efficient. Actually, we could "invert the sense" of the vector, and have it store a 1 if the thing is **not** present....call it `clean_bit_vec`...